packages/dd-trace/src/opentelemetry/metrics/otlp_transformer.js

```diff
@@ -0,0 +1,280 @@
+   * @param {boolean} isJson - Whether to format for JSON output
+   * @returns {Object[]} Array of scope metric objects
+   * @private
+   */
+  #transformScope (metrics, isJson = false) {
+    const groupedMetrics = this.groupByInstrumentationScope(metrics)
+    const scopeMetrics = []
+
+    for (const [, metricsInScope] of groupedMetrics) {
+      const firstMetric = metricsInScope[0]
+      const instrumentationScope = firstMetric.instrumentationScope || {}
+      const { name = '', version = '', schemaUrl = '', attributes = {} } = instrumentationScope
+
+      const scope = {
+        name,
+        version,
+        droppedAttributesCount: 0
+      }
+
+      if (attributes) {
+        const transformed = isJson ? this.attributesToJson(attributes) : this.transformAttributes(attributes)
+        if (transformed.length) {
+          scope.attributes = transformed
+        }
+      }
+
+      scopeMetrics.push({
+        scope,
+        schemaUrl,
+        metrics: metricsInScope.map(metric =>
+          isJson ? this.#transformMetricToJson(metric) : this.#transformMetric(metric)
+        )
+      })
+    }
+
+    return scopeMetrics
+  }
+
+  /**
+   * Transforms a single metric to protobuf format.
+   * @private
+   */
+  #transformMetric (metric) {
+    const result = {
+      name: metric.name,
+      description: metric.description || '',
+      unit: metric.unit || ''
+    }
+
+    const temporality = metric.temporality === TEMPORALITY.CUMULATIVE
+      ? AGGREGATION_TEMPORALITY_CUMULATIVE
+      : AGGREGATION_TEMPORALITY_DELTA
+
+    switch (metric.type) {
+      case METRIC_TYPES.HISTOGRAM:
+        result.histogram = {
+          dataPoints: metric.data.map(dp => ({
+            attributes: this.transformAttributes(dp.attributes),
+            startTimeUnixNano: dp.startTimeUnixNano,
+            timeUnixNano: dp.timeUnixNano,
+            count: dp.count,
+            sum: dp.sum,
+            bucketCounts: dp.bucketCounts || [],
+            explicitBounds: dp.explicitBounds || [],
+            min: dp.min,
+            max: dp.max
+          })),
+          aggregationTemporality: temporality
+        }
+        break
+
+      case METRIC_TYPES.COUNTER:
+      case METRIC_TYPES.OBSERVABLECOUNTER:
+      case METRIC_TYPES.UPDOWNCOUNTER:
+      case METRIC_TYPES.OBSERVABLEUPDOWNCOUNTER:
+        result.sum = {
+          dataPoints: metric.data.map(dp => this.#transformNumberDataPoint(dp)),
+          aggregationTemporality: temporality,
+          isMonotonic: metric.type === METRIC_TYPES.COUNTER || metric.type === METRIC_TYPES.OBSERVABLECOUNTER
+        }
+        break
+
+      case METRIC_TYPES.GAUGE:
+        result.gauge = {
+          dataPoints: metric.data.map(dp => this.#transformNumberDataPoint(dp))
+        }
+        break
+    }
+
+    return result
+  }
+
+  /**
+   * Transforms a single metric to JSON format.
+   * @private
+   */
+  #transformMetricToJson (metric) {
```

Comment:
This method could be combined with the transformMetric method. I would just add another argument to the method and there seem to be three different spots where different methods must be called depending on a boolean argument about it being a json or not (one of these would a new method that abstracts the part for METRIC_TYPES.HISTOGRAM).